src/erlcloud_aws.erl

```diff
@@ -574,6 +590,42 @@ prop_to_list_defined( Name, Props ) ->
     end.
 
 
+-spec get_role_credentials(aws_config()) -> {ok, #role_credentials{}} | {error, term()}.
+get_role_credentials(Config) ->
+    case application:get_env(erlcloud, role_credentials) of
+        {ok, #role_credentials{expiration_gregorian_seconds = Expiration} = Credentials} ->
+            Now = calendar:datetime_to_gregorian_seconds(calendar:universal_time()),
+            %% Get new credentials if these will expire in less than 5 minutes
+            case Expiration - Now < 300 of
+                true -> get_credentials_from_role(Config);
+                false -> {ok, Credentials}
+            end;
+        undefined ->
+            get_credentials_from_role(Config)
+    end.
+
+-spec get_credentials_from_role(aws_config()) -> {ok, #role_credentials{}} | {error, term()}.
+get_credentials_from_role(#aws_config{assume_role = AssumeRoleOptions} = Config) ->
+    RoleArn = proplists:get_value(role_arn, AssumeRoleOptions, undefined),
+    SessionName = proplists:get_value(session_name, AssumeRoleOptions,
+                                      "erlcloud"),
+    SessionExpiration = proplists:get_value(session_expiration,
+                                            AssumeRoleOptions, 900),
+    case catch erlcloud_sts:assume_role(Config, RoleArn, SessionName,
+                                        SessionExpiration) of
+        {#aws_config{}=_NewConfig, Creds} ->
+            Record = #role_credentials{
+                access_key_id =  proplists:get_value(access_key_id, Creds),
+                secret_access_key = proplists:get_value(secret_access_key, Creds),
+                session_token = proplists:get_value(session_token, Creds),
+                expiration_gregorian_seconds = timestamp_to_gregorian_seconds(
+                    proplists:get_value(expiration, Creds))},
+            application:set_env(erlcloud, role_credentials, Record),
```

Comment:
Library already uses env to cache metadata_credentials. I see your point of multiple roles being assumed,  we can cache in a "dictionary" based in the role_arn.  The idea is to allow to a assume a role  per request and once you assume a role the library will cache it for an amount of time per assumed role. This is different from the role_arn in the profile file.

This use case is  for applications that run in ec2 instances and need to assume a role and do not have the profile file, just the policy attached to the IAM Role of the instance. It's true that you may have a cascade of assume role but you do not have a good way of express that to be used in the library. My Idea is to keep it simple and do it in a per request base where you can change the assume_role (not implemented yet)